cmd/agent/Dockerfile.windows

```diff
@@ -0,0 +1,18 @@
+FROM grafana/agent-build-image:latest-windows as builder
```

Comment:
We should use the explicit version here instead of `latest` for reproducible builds (also we don't push a `latest` tag anyway)

Reply:
I will update. Just as an fyi, something is pushing a latest tag: https://hub.docker.com/layers/grafana/agent/latest/images/sha256-b462aedb7a3540f13655b782d3514705a62231d1834cb09fed5401c17f1b0637?context=repo

Reply:
Yes, there's a latest tag for `grafana/agent` but not `grafana/agent-build-image` :)

Reply:
check that, that is for the agent docker image not the agent-build-image